src/utils/read-file.ts

```diff
@@ -0,0 +1,44 @@
+import { readFileSync } from "fs";
+import { User } from "@src/types";
+import { logger } from "@src/commons/logger";
+
+/**
+ * Read file and return list of users
+ * @param {string} path - File name
+ * @returns {User[] | undefined} List of users
+ */
+export function readUserDataFile(path: string): User[] | undefined {
+  try {
+    const fileData = readFileSync(path, "utf-8");
+    // If file data is empty, log file empty error and return undefined
+    if (fileData === "") {
+      logger.error("File empty!");
+      return;
+    }
+    // Convert file data to string array, array item is one line in file
+    const userLines: string[] = fileData.split("\n");
+    // Init hasError variable to check format data in userLines
+    let hasError: boolean = false;
+    const users: User[] | undefined = userLines.map((userLine) => {
+      // Convert userLines to userData use split(" ")
+      const userData: string[] = userLine.split(" ");
+      // If splitted data of items different 3, log incorrect format
+      if (userData.length !== 3) {
+        logger.error("Data format incorrect!");
+        hasError = true;
```

Comment:
logic chỗ này sao sao nhỉ nếu data không hợp lệ thì vẫn trả về à em? nếu vẫn trả về thì chỗ log anh nghĩ là log warning thôi còn nếu đã coi là error thì skip qua line này không trả về và log error chi tiết hơn chút để người dùng biết line nào bị lỗi nữa thì hợp lý hơn

Reply:
chỗ này đang ở trong vòng map, khi có lỗi em k break ra được luôn nên e cắm 1 cái flag là hasError để check,
em đã update thêm line gây ra lỗi rồi ạ.

Reply:
nếu muốn break ra thì em dùng vòng for được mà (vd for of). Chỗ này nếu data lỗi thì anh thấy em vẫn push vào mảng users, tuy nhiên ở L42 nếu hasError = true thì return luôn thì cái việc push vào mảng users của em ở trên hơi vô nghĩa. chỗ này không cần update cũng được mà anh note lại để em để ý kĩ logic khi làm thôi